include/ur_client_library/comm/pipeline.h

```diff
@@ -328,7 +328,15 @@ class Pipeline
    */
   bool getLatestProduct(std::unique_ptr<T>& product, std::chrono::milliseconds timeout)
   {
-    return queue_.waitDequeTimed(product, timeout);
+    // If the queue has more than one package, get the latest one.
+    bool res = false;
+    while (queue_.tryDequeue(product))
+    {
+      res = true;
+    }
+
+    // If the queue is empty, wait for a package.
+    return res || queue_.waitDequeTimed(product, timeout);
```

Comment:
Personally not a fan of this type of return structure, but I can accept that as subjective.